tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -265,29 +285,43 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
       instr = gemm_call;
     }
 
-    // Attempt to elide an FP8 GEMM with scaled inputs as described by steps 1
-    // through 3 detailed above and rewrite into a Custom Call.
-    if (Match(
-            instr,
-            m::CustomCall(
-                {kCublasLtMatmulCallTarget},
-                m::AnyOf<HloInstruction>(
-                    OptionalBitcast(
-                        &a_bitcast,
-                        m::MultiplyAnyOrder(&a_binary, m::Convert(m::Op(&a)),
-                                            m::Broadcast(m::Op(&a_scale)))),
-                    OptionalBitcast(&a_bitcast,
-                                    m::Divide(&a_binary, m::Convert(m::Op(&a)),
-                                              m::Broadcast(m::Op(&a_scale))))),
-                m::AnyOf<HloInstruction>(
-                    OptionalBitcast(
-                        &b_bitcast,
-                        m::MultiplyAnyOrder(&b_binary, m::Convert(m::Op(&b)),
-                                            m::Broadcast(m::Op(&b_scale)))),
-                    OptionalBitcast(
-                        &b_bitcast,
-                        m::Divide(&b_binary, m::Convert(m::Op(&b)),
-                                  m::Broadcast(m::Op(&b_scale)))))))) {
+    if (Match(instr,
+              m::CustomCall(
+                  {kCublasLtMatmulCallTarget},
+                  m::AnyOf<HloInstruction>(
+                      OptionalBitcast(
+                          &a_bitcast,
+                          OptionalCopy(&a_copy,
+                                       OptionalReshapeOrBitcast(
```

Comment:
I think you'll have to pass `a_bitcast2` to `CreateF8CustomCall`, and take it into account when creating the custom call (probably by moving the rehshape/bitcast to run before the convert op), as is done for `a_bitcast`.

Reply:
![image](https://user-images.githubusercontent.com/25590028/221945346-34513d1c-81a0-4a69-87be-ba9445531d99.png) and 
![image](https://user-images.githubusercontent.com/25590028/221945455-7ad00f2f-afb8-434c-8632-27b4fff049bf.png)
Please see the HLO before and after gemm_rewrite.a_bitcast2 will be the bitcast.11 which is of shape [4,512,32,4]. If passed into CreateF8CustomCall, the shape will be not reasonable. Could you elaborate on "by moving the rehshape/bitcast to run before the convert op"? Reshape/Bitcast are in-place before gemm_rewrite get called.

Reply:
In your example, you don't need to move `a_bitcast2` to be before the convert op, but in general, you might. If `a_bitcast2` is present, but `a_bitcast` is not, then simply removing `a_bitcast2` is not safe, because `a_bitcast2` might change the shape from A to B, and the `dot` operation expects shape B, not A.

In your example, `a_bitcast` also changes the shape, so `a_bitcast2` can be safely removed.

If you want, I can write up an HLO example that this PR will fail on.